static/app/components/charts/utils.tsx

```diff
@@ -180,6 +181,23 @@ export function getSeriesApiInterval(datetimeObj: DateTimeObject) {
   return '1h';
 }
 
+export type GranularityStep = [number, string];
+export type GranularityLadder = GranularityStep[];
+
+/**
+ * Given a duration in minutes and a `GranularityLadder` object picks the correct granularity at that duration and return the interval as a string e.g. `"10m"`
+ */
+export function findGranularityIntervalForMinutes(
+  minutes: number,
+  granularities: GranularityLadder
+) {
+  const step = granularities.find(([threshold]) => {
+    return minutes >= threshold;
+  }) as GranularityStep; // This can never be undefined, because the first step is 0, so any duration has to be larger than at least the first
```

Comment:
l: should we add code that does an invariant check here? maybe log error to Sentry?

Reply:
I was wondering how to best tackle this problem overall. One thing I considered is making some kind of `createGranularityLadder` function that would make it impossible to create an invalid ladder 🤔 

For the invariant check, did you mean validating the ladder before using, or a fallback, or something else?

Reply:
validating that `granularities.find` is defined and throw an error it is not.

Reply:
@AbhiPrasad what do you think of https://github.com/getsentry/sentry/pull/60278/commits/d360458ae6befc087c63935d29120f5e8a3ffb9d? I like it more, since it makes it _impossible_ set up an incorrect granularity latter. Plus, the type cast is now colocated with the logic that makes it impossible. Since granularities are actually static, this seems way better. No more primitive obsession!

Reply:
ooo nice, a true parse don't validate common W